haystack/dataclasses/chat_message.py

```diff
@@ -87,7 +91,70 @@ class TextContent:
     text: str
 
 
-ChatMessageContentT = Union[TextContent, ToolCall, ToolCallResult]
+@dataclass
+class ImageContent:
+    """
+    The image content of a chat message.
+    """
+
+    base64_image: str
+    mime_type: Optional[str] = None
+    meta: Dict[str, Any] = field(default_factory=dict)
+    provider_options: Dict[str, Any] = field(default_factory=dict)
+    # too many parameters?
+
+    def __repr__(self) -> str:
+        """
+        Return a string representation of the ImageContent, truncating the base64_image to 100 bytes.
+        """
+        fields = []
+        truncated_data = self.base64_image[:100] + "..." if len(self.base64_image) > 100 else self.base64_image
+        fields.append(f"base64_image={truncated_data!r}")
+        fields.append(f"mime_type={self.mime_type!r}")
+        fields.append(f"meta={self.meta!r}")
+        fields.append(f"provider_options={self.provider_options!r}")
+        fields_str = ", ".join(fields)
+        return f"{self.__class__.__name__}({fields_str})"
+
+    @classmethod
+    def from_file_path(
+        cls,
+        file_path: str,
+        mime_type: Optional[str] = None,
+        meta: Optional[Dict[str, Any]] = None,
+        provider_options: Optional[Dict[str, Any]] = None,
+    ) -> "ImageContent":
```

Comment:
I realize this is a WIP, but just wanted to give some feedback:
* We may want to make the utf-8 configurable to the user
* Could be worth wrapping the file loading in try-except like we do for our Converters